### PR TITLE
session3 assignment

### PR DESCRIPTION
**Pull request by miguelrefuerzo**
session3 assignment

# Media

Get:
image

Post:
image

Put:
image

Get{id}:
image

Delete:
image

### CODE REVIEW

File: Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+        public ProductService(AppDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<List<ProductDto>> GetProducts()
+        {
+            // TODO get products
+            var product = await _context.Products.ToListAsync();
```
Comment (laira-salvador) [inline Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs]: You can chain .Select method after '_context.Products' then call the execute command. In this way we can be sure to run the command in the database and not in the client.

File: Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            return product.Select(p => new ProductDto(p.Id, p.Name, p.ShopId)).ToList();
+        }
+
+        public async Task<ProductDto> GetProduct(int id)
+        {
+            // TODO get product
+            var product = await _context.Products.FindAsync(id);
```
Comment (laira-salvador) [inline Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs]: You can use .Select method to directly cast the Product to ProductDto

**Review by laira-salvador — CHANGES_REQUESTED** [review]
- Use Select method to pick properties and cast it to another type

File: Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs (hunk 2 above)
Comment (miguelrefuerzo) [reply Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs]: hello here is the updated code
image

File: Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs (hunk 1 above)
Comment (miguelrefuerzo) [reply Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs]: hello here is the updated code
image

**Review by laira-salvador — APPROVED** [review]

**Pull request merged**